src/frontend/bibletime.cpp

```diff
@@ -332,3 +335,39 @@ void BibleTime::openSearchDialog(BtConstModuleList modules,
 }
 
 void BibleTime::openFindWidget() { m_findWidget->showAndSelect(); }
+
+#ifdef BUILD_TEXT_TO_SPEECH
+void BibleTime::speakText(QString const & text) {
+    if (!m_textToSpeech)
+        m_textToSpeech = createTextToSpeechInstance();
+
+    m_textToSpeech->say(text);
+}
+
+std::unique_ptr<QTextToSpeech> BibleTime::createTextToSpeechInstance() {
+    std::unique_ptr<QTextToSpeech> tts;
+
+    // restore settings from config
+    QString const configuredEngine = btConfig().value<QString>(QStringLiteral("GUI/ttsEngine"));
```

Comment:
As I [explained](https://github.com/bibletime/bibletime/pull/356#pullrequestreview-2660364711) earlier, I think it were better to use `auto const` here and similarly in other places like this. I think it is relatively safe to use `auto` because it is unlikely that the type of expression will change in ways which could cause silent run-time failures, especially the return value of `BtConfig::value<QString>()`. But if you have a different strong opinion on the use of `auto`, I'd be interested in learning about your perspective. :)

For other places I'd like to see `auto` being used, I'll just comment briefly by stating the expected declaration, e.g. `auto &` or `auto const` or similar, so you can more easily spot those places.

Reply:
Thanks for the explicit suggestions, which I've implemented in the newly pushed version.

> But if you have a different strong opinion on the use of auto, I'd be interested in learning about your perspective. :)

While I personally have some preference for explicitly specifying the type (except for types like iterators - `std::vector<SomeClass>::iterator` etc.) as I personally find code a bit more readable/obvious then (type is clear at first glance, while it might need a second/glance thought when it's `auto`), I don't have a strong opinion here on this and code style in general and try to adhere to whatever the code style of the particular project I'm contributing to is using (and am happy to change it when I missed anything at first).